lib/pavilion/plugins/commands/wait.py

```diff
@@ -40,17 +42,25 @@ def _setup_arguments(self, parser):
                  'test IDs and series IDs.  If no value is provided, the most '
                  'recent series submitted by this user is checked.'
         )
+        parser.add_argument(
+            '-s', '--silent', action='store_true'
+        )
 
     def run(self, pav_cfg, args):
-        # Store the initial time for timeout functionality.
         start_time = time.time()
 
         tmp_statuses = status.get_statuses(pav_cfg, args, self.errfile)
 
         final_statuses = 0
 
-        while (final_statuses < len(tmp_statuses)) and \
-              ((time.time() - start_time) < float(args.timeout)):
```

Comment:
I argued with Nick about doing it this way. The status is for informational purposes only. The list of statuses is subject to change, and is generally not reliable. It even says so in their documentation.

The right way to tell if a test is done isn't to check its status, it's to check use the test.wait() method. It's a slightly harder problem to do this for multiple tests, but not bad. I'd appreciate it if you'd fix it the right way. 

Sorry I didn't tell you this in the first place. I forgot how it was implemented.

Reply:
If I use the `test.wait()` method, then I won't be able to periodically print the test statuses while waiting for the results. 

Instead of checking the statuses, I think I'll check for the `RUN_COMPLETE` file, which is how `test.wait` checks if a test is finished.